src/typescript/azure-openapi-validator/index.ts

```diff
@@ -7,18 +7,23 @@ import { Message } from "../jsonrpc/types";
 import { rules, OpenApiTypes, MergeStates } from "./rule";
 
 // register rules
-require("./rules/DescriptionMustNotBeNodeName");
-require("./rules/ControlCharactersAreNotAllowed");
-require("./rules/ArraySchemaMustHaveItems");
-require("./rules/PostOperationIdContainsUrlVerb");
-require("./rules/LicenseHeaderMustNotBeSpecified");
+let fs = require('fs');
+let path = require('path');
+
+for (const ruleFile of fs.readdirSync(path.join(__dirname, "rules"))) {
```

Comment:
I'm not a big fan of implicit discovery -- and this causes trouble with some work I'm doing with building a static-linker for nodejs.

As much as I can appreciate the desire to not have to have an explicit reference to the individual rule files, I'd prefer that over using `readdirSync` and arbitrarily using `require` on the discovered files.

Reply:
The problem with non-implicit loading in this case is that there are up to 28(!) additional rules requested by various teams (if none of them are combined, and are all viable automated checks.) These could be placed in a single file for re-export to the `run` method, but the problem there comes from two sources:

1. Node's resolution at runtime problem (no static analysis to see if a referenced import exists)
2. The extremely large number of manually managed imports would make it easy to lose track of them.

I'm curious about your thoughts given these constraints and challenges. If we just need to suck it up and manually manage everything because that's the required style for the project, that's fine.

Reply:
Perhaps the halfway point would be a build-time script that discovers them, and generates a single file that `require`s them all. Then just require that one file and be done with it.

That way, you can make it not a manual process, and we can still target the validator for static-linking.

Reply:
Good suggestion - I'll make this change.